src/NServiceBus.AcceptanceTests/Sagas/When_saga_id_changed.cs

```diff
@@ -26,10 +26,9 @@ await Scenario.Define<Context>()
                 .ExpectFailedMessages();
 
             Assert.AreEqual(1, exception.FailedMessages.Count);
-            Assert.AreEqual(((Context) exception.ScenarioContext).MessageId, exception.FailedMessages.Single().MessageId, "Message should be moved to errorqueue");
-
-            Assert.True(exception.ScenarioContext.LoggedExceptions.Any(e =>
-                e.Message.Contains("A modification of IContainSagaData.Id has been detected. This property is for infrastructure purposes only and should not be modified. SagaType: " + typeof(Endpoint.SagaIdChangedSaga))));
+            var failedMessage = exception.FailedMessages.Single();
+            Assert.AreEqual(((Context) exception.ScenarioContext).MessageId, failedMessage.MessageId, "Message should be moved to errorqueue");
+            StringAssert.Contains("A modification of IContainSagaData.Id has been detected. This property is for infrastructure purposes only and should not be modified. SagaType:", failedMessage.Exception.Message);
```

Comment:
How about using `Assert.That` here as well like in https://github.com/Particular/NServiceBus/pull/3768/files#diff-748e58145169da43f2528a43ee6a7d64R77